src/modules/projects/components/tables/ProjectClientEnrollmentsTable.tsx

```diff
@@ -40,65 +45,135 @@ const isHouseholdWithStaff = (
+
+  return (
+    <Box aria-label={allNames.join(', ')}>
+      {first}{' '}
+      {rest.length > 0 && (
+        <Tooltip arrow title={rest.join(', ')}>
+          <Chip sx={{ mb: 0.5 }} size='small' label={`+${rest.length} more`} />
+        </Tooltip>
+      )}
+    </Box>
+  );
+};
+
+export const HOUSEHOLD_ASSIGNED_STAFF_COL: GenericTableWithDataColumnDef<
+  HouseholdWithStaffAssignmentsFragment,
+  EnrollmentQueryVariables
+> = {
+  header: 'Assigned Staff',
+  optional: {
+    defaultHidden: true,
+    queryVariableField: 'includeStaffAssignment',
+  },
+  key: 'assigned_staff',
+  render: renderHouseholdAssignedStaff,
+};
+
 export const ENROLLMENT_COLUMNS: {
-  [key: string]: ColumnDef<
-    | ClientEnrollmentFieldsFragment
+  [key: string]: GenericTableWithDataColumnDef<
+    | ClientEnrollmentTableFields
     | ProjectEnrollmentQueryEnrollmentFieldsFragment
+    | ClientEnrollmentFieldsFragment,
+    EnrollmentQueryVariables
   >;
 } = {
   entryDate: {
     header: 'Entry Date',
+    key: 'entryDate',
     render: (e) => (
       <DateWithRelativeTooltip dateString={e.entryDate} preciseTime={false} />
     ),
   },
   exitDate: {
     header: 'Exit Date',
+    key: 'exitDate',
+    optional: {
+      defaultHidden: true,
+      // queryVariableField not provided here, since we need to fetch exitDate anyway in order to show the status
+      // and correctly aria-label the row action
+    },
     render: (e) => {
-      if (e.exitDate)
+      if (e.exitDate) {
         return (
           <DateWithRelativeTooltip
             dateString={e.exitDate}
             preciseTime={false}
           />
         );
+      }
     },
   },
   enrollmentStatus: {
     header: 'Status',
+    key: 'status',
     render: (e) => <EnrollmentStatus enrollment={e} />,
   },
-};
-
-export const ASSIGNED_STAFF_COL = {
-  header: 'Assigned Staff',
-  optional: true,
-  defaultHidden: true,
-  key: 'assigned_staff',
-  render: (hh: HouseholdWithStaffAssignmentsFragment) => {
-    if (!hh.staffAssignments?.nodes.length) return;
-
-    const allNames = hh.staffAssignments.nodes.map(
-      (staffAssignment) => staffAssignment.user.name
-    );
-
-    const first = allNames[0];
-    const rest = allNames.slice(1);
-
-    return (
-      <Box aria-label={allNames.join(', ')}>
-        {first}{' '}
-        {rest.length > 0 && (
-          <Tooltip arrow title={rest.join(', ')}>
-            <Chip
-              sx={{ mb: 0.5 }}
-              size='small'
-              label={`+${rest.length} more`}
-            />
-          </Tooltip>
-        )}
-      </Box>
-    );
+  moveInDate: {
+    header: 'Move-in Date',
+    key: 'moveInDate',
+    optional: {
+      defaultHidden: true,
+      queryVariableField: 'includeMoveInDate',
+    },
+    render: (e) => {
+      if (e.moveInDate) {
+        return (
+          <DateWithRelativeTooltip
+            dateString={e.moveInDate}
+            preciseTime={false}
+          />
+        );
+      }
+    },
```

Comment:
nit: avoiding extra (one-char) enrollment var makes it more succinct
```suggestion
    render: ({ moveInDate }) =>
      moveInDate && (
        <DateWithRelativeTooltip dateString={moveInDate} preciseTime={false} />
      ),
```